nopassword/models.py

```diff
@@ -64,7 +64,9 @@ def create_code_for_user(cls, user, next=None):
         if not user.is_active:
             return None
 
-        code = cls.generate_code(length=getattr(settings, 'NOPASSWORD_CODE_LENGTH', 20))
+        code = None
+        while not cls.code_is_valid(code):
```

Comment:
Sounds sensible, if the random number generator somehow degrades then it might only provide a single code. Ideally it should just raise and an admin can look deeper to why no codes are available.